cc-proxy/src/mergeConversations.js

```diff
@@ -0,0 +1,143 @@
+const _ = require('lodash');
+const RestError = require('./restError');
+const { authorize, getTagsAndRoles } = require('./proxy/permissions');
+const raisely = require('./raiselyRequest');
+
+// taken from reports.js
+
+/**
+ * Count the number of attendees, defined as hosts, co-hosts and guests
+ * removes duplicates (eg if someone were a host of one conversation
+ * and then a guest)
+ * @param {*} rsvps
+ * @returns {integer}
+ */
+function countAttendees(rsvps) {
+	const attendeeTypes = ['host', 'co-host', 'guest'];
+	const allAttendees = rsvps
+		.filter((r) => attendeeTypes.includes(r.type))
+		.map((r) => r.userUuid);
+	const uniqueAttendees = _.uniq(allAttendees);
+	return uniqueAttendees.length;
+}
+
+// compares both conversations
+// returns Uuid with more RSVPs
+async function mostRSVPs(conversationUuid1, conversationUuid2) {
+	let conversation1RSVP = await raisely(
+		{
+			method: 'GET',
+			path: `/events/${conversationUuid1}/rsvps?private=1`,
+			escalate: true,
+		},
+		req
+	);
+	let conversation2RSVP = await raisely(
+		{
+			method: 'GET',
+			path: `/events/${conversationUuid2}/rsvps?private=1`,
+			escalate: true,
+		},
+		req
+	);
+
+	let conversation1AttendeesCount = countAttendees(conversation1RSVP.data);
+	let conversation2AttendeesCount = countAttendees(conversation2RSVP.data);
+
+	return conversation1AttendeesCount >= conversation2AttendeesCount
+		? conversationUuid1
+		: conversationUuid2;
+}
+
+async function mergeConversations(req) {
+	let { conversationUuid1, conversationUuid2 } = req.body.data;
+
+	let conversation1Data = await raisely(
+		{
+			method: 'GET',
+			path: `/events/${conversationUuid1}?private=1`,
+			escalate: true,
+		},
+		req
+	);
+
+	let conversation2Data = await raisely(
+		{
+			method: 'GET',
+			path: `/events/${conversationUuid2}?private=1`,
+			escalate: true,
+		},
+		req
+	);
+
+	// Sanity check: should reject merge if conversation1Data.name !== conversation2Data.name
+	if (conversation1Data.data.name !== conversation2Data.data.name) {
+		console.error('Both conversations have the same name!');
+	}
+
+	// select which uuid to keep. Keep the uuid with the most RSVPs
+	const uuidToKeep = await mostRSVPs(conversationUuid1, conversationUuid2);
+
+	let conversationToKeep =
+		conversation1Data.data.uuid == uuidToKeep
+			? conversation1Data
+			: conversation2Data;
+	let conversationToDelete =
+		conversation1Data.data.uuid !== uuidToKeep
+			? conversation2Data
+			: conversation1Data;
+
+	// go through all the keys in the conversation to keep the
+	// and merge them into the selected conversation
+	// note that some fields only appear after they are completed (eg cash donations)
+	// (at key, if it has values, keep the field.
+	// if not, check the difference, which field to keep
+
+	for (let [key, value] of Object.entries(conversationToKeep)) {
```

Comment:
This should be `conversationToDelete` 
They may not have the exact same fields populated, if you don't iterate over the keys of conversationToDelete, you may miss some of it's keys.

Reply:
This will do a shallow copy which will lose all the custom fields from the conversationToDelete.
When `key === 'private'` or `'public'` it will see that there is already a value (the object) there and will 
You'll need to do a deeper copy for public and private.
You can use the object spread short hand for public & private:
```
conversationToKeep.public = {
   ...(conversationToDelete.public || {}),
   ...(conversationToKeep.public || {})
};
```